src/Microsoft.TemplateEngine.Cli/Commands/TemplateGroupCommand.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.CommandLine.Help;
+using System.CommandLine.Invocation;
+using Microsoft.TemplateEngine.Abstractions;
+
+namespace Microsoft.TemplateEngine.Cli.Commands
+{
+    internal class TemplateGroupCommand : Command, ICommandHandler
+    {
+        private IEngineEnvironmentSettings _environmentSettings;
+        private ITemplateInfo _template;
+        private NewCommand _newCommand;
+        private Dictionary<string, Option> _templateSpecificOptions;
+
+        public TemplateGroupCommand(NewCommand newCommand, IEngineEnvironmentSettings environmentSettings, ITemplateInfo template) : base(template.ShortNameList[0], template.Name + Environment.NewLine + template.Description)
+        {
+            _newCommand = newCommand;
+            _environmentSettings = environmentSettings;
+            foreach (var item in template.ShortNameList.Skip(1))
+            {
+                AddAlias(item);
+            }
+            _templateSpecificOptions = TemplateGroupArgs.AddToCommand(this, template);
```

Comment:
probably we want to move this logic to command itself, as with other Args classes